File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-33b-instruct/Avatar/Python/Java/codeforces_534_B.java
Here is the Java version of the Python code:

```java
import java.util.Scanner;

public class Main {
    static int v1, v2, t, d, a, v, distance;
    static final int FirstCondition = 989, SecondCondition = 711, OuterLoopLimit = 748, OuterLoopStep = 747;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        v1 = scanner.nextInt();
        v2 = scanner.nextInt();
        t = scanner.nextInt();
        d = scanner.nextInt();
        a = 1;
        v = v1;
        distance = 0;

        for (int LoopIndexOut = 0; LoopIndexOut < OuterLoopLimit / OuterLoopStep; LoopIndexOut++) {
            recursiveVelocityCalculator(0, t, 1);
        }

        System.out.println(distance);
    }

    static void recursiveVelocityCalculator(int i, int stop, int step) {
        if (step == 0 || (step > 0 && i >= stop) || (step < 0 && i <= stop)) {
            return;
        }
        if ((FirstCondition & SecondCondition) != 0) {
            if (v - v2 > (t - i - 1) * d) {
                a *= -1;
                v = (t - i - 1) * d + v2;
            }
        }
        distance += v;
        v += a * d;
        recursiveVelocityCalculator(i + step, stop, step);
    }
}
```

Please note that the Java version of the code is slightly different from the Python version. The Python code uses global variables, while the Java version uses static variables. Also, the Python code uses the `input()` function to get the input, while the Java version uses the `Scanner` class to get the input.

File: dataset/Intermediate/Translation/Avatar/v2-Python-code/codeforces_534_B.py
(v1, v2) = input().split()
(t, d) = input().split()
v1 = int(v1)
v2 = int(v2)
t = int(t)
d = int(d)
a = 1
v = v1
distance = 0
FirstCondition = 989
SecondCondition = 711
OuterLoopLimit = 748
OuterLoopStep = 747
for LoopIndexOut in range(OuterLoopLimit // OuterLoopStep):

    def recursive_velocity_calculator(i, stop, step):
        global v, distance, a
        if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
            return
        if FirstCondition & SecondCondition:
            if v - v2 > (t - i - 1) * d:
                a *= -1
                v = (t - i - 1) * d + v2
        distance = distance + v
        v += a * d
        recursive_velocity_calculator(i + step, stop, step)
    recursive_velocity_calculator(0, t, 1)
print(distance)